### PR TITLE
Update toplayer visiblity on workspace switch

### PR DESCRIPTION
**Pull request by Consolatis**
Fixes:
- #1158

Only had a very short test nested.

I am also not sure if `view.c` is actually the correct place for the new `view_update_top_layer()` function:
- As it depends on a set of views it could also belong to `desktop.c`.
- As it toggles the visibility of a specific layershell layer it could also belong to `layers.c`.

### CODE REVIEW

File: src/view.c [hunk 1]
```diff
@@ -1097,6 +1102,33 @@ view_toggle_fullscreen(struct view *view)
 	view_set_fullscreen(view, !view->fullscreen);
 }
 
+void
+view_update_top_layer(struct server *server)
+{
+	struct view *view;
+	struct output *output;
+
+	uint32_t top = ZWLR_LAYER_SHELL_V1_LAYER_TOP;
```
Comment (Consolatis) [inline src/view.c:1111]: remove empty line

**Comment by jlindgren90** [comment]
Didn't test but LGTM. It looks like this would also correctly handle the case where you have two fullscreen windows and close one of them. :+1:

**Comment by johanmalm** [comment]
LGTM. Have tested and seems to work great.

Nice to increase usage of `for_each_view()`

Agree, `desktop.c` feel like a better home. Even if we're dragging some layer-stuff it's all a function of the views.

Suggest `s/view_update_top_layer/desktop_update_top_layer_visibility/`

**Comment by Consolatis** [comment]
> Agree, `desktop.c` feel like a better home. Even if we're dragging some layer-stuff it's all a function of the views.

Moved to `desktop.c`

> Suggest `s/view_update_top_layer/desktop_update_top_layer_visibility/`

Changed

**Comment by johanmalm** [comment]
Really nice patch. Thanks!

**Pull request merged**